client-next/src/components/MapView/MapView.tsx

```diff
@@ -29,20 +36,31 @@ export function MapView({
   loading: boolean;
 }) {
   const onMapDoubleClick = useMapDoubleClick({ tripQueryVariables, setTripQueryVariables });
-  const [showPopup, setShowPopup] = useState<LngLat | null>(null);
+  const [showContextPopup, setShowContextPopup] = useState<LngLat | null>(null);
+  const [showPropsPopup, setShowPropsPopup] = useState<PopupData | null>(null);
 
   return (
     <div className="map-container below-content">
       <Map
         // @ts-ignore
         mapLib={import('maplibre-gl')}
         // @ts-ignore
-        mapStyle={mapStyle}
+        mapStyle="http://localhost:8080/otp/routers/default/inspector/vectortile/style.json"
         initialViewState={initialViewState}
         onDblClick={onMapDoubleClick}
         onContextMenu={(e) => {
-          setShowPopup(e.lngLat);
+          setShowContextPopup(e.lngLat);
         }}
+        interactiveLayerIds={['regular-stop']}
+        onClick={(e) => {
+          const props = e.features[0];
```

Comment:
> Error: src/components/MapView/MapView.tsx(56,25): error TS18048: 'e.features' is possibly 'undefined'.

You can't access a property of an object that could potentially be `undefined`, you need to check if it is undefined first (like a null check). Surrounding the lines that access `e` with a conditional block would solve it: